walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,119 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> conflictingTimes = findConflictingTimes(events, request.getAttendees());
+    List<TimeRange> possibleTimes = findPossibleTimes(conflictingTimes, request.getDuration());
+
+    return possibleTimes;
+  }
+
+  public List<TimeRange> findConflictingTimes(Collection<Event> events,
```

Comment:
Double check whether all the methods need to be declared as public. Make each class or member as inaccessible as possible.  Aside from the classes and members you are intentionally designing to be exposed as a public API, all remaining classes and members should use the lowest visibility possible.

See also Effective Java Chapter 4: A well-designed
component hides all its implementation details, cleanly separating its API from its
implementation. Components then communicate only through their APIs and are
oblivious to each others’ inner workings.